File: app/src/main/java/com/example/android/pingpongscorekeeper/components/PingPongPlayer.java
package com.example.android.pingpongscorekeeper.components;

public class PingPongPlayer
{
    private final String name;
    private int currentSetScore;
    private int numberOfSetsWon;

    public PingPongPlayer(String name)
    {
        this.name = name;
        currentSetScore = 0;
        numberOfSetsWon = 0;
    }

    public void resetNumberOfSetsWon()
    {
        numberOfSetsWon = 0;
    }

    public void resetCurrentSetScore()
    {
        currentSetScore = 0;
    }

    public int getCurrentSetScore()
    {
        return currentSetScore;
    }

    public int getNumberOfSetsWon()
    {
        return numberOfSetsWon;
    }

    public void incrementCurrentSetScore()
    {
        currentSetScore++;
    }

    public void incrementNumberOfSetsWon()
    {
        numberOfSetsWon++;
    }

    public String getName() {
        return name;
    }
}

File: app/src/main/java/com/example/android/pingpongscorekeeper/components/PingPongPlayer.kt
package com.example.android.pingpongscorekeeper.components

class PingPongPlayer(var name: String) {

    var currentSetScore: Int = 0
        private set

    var numberOfSetsWon: Int = 0
        private set

    fun resetNumberOfSetsWon() {
        numberOfSetsWon = 0
    }

    fun resetCurrentSetScore() {
        currentSetScore = 0
    }

    fun reset() {
        resetNumberOfSetsWon()
        resetCurrentSetScore()
    }

    fun incrementCurrentSetScore() = currentSetScore++

    fun incrementNumberOfSetsWon() = numberOfSetsWon++
}
